sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOReadTest.java

```diff
@@ -617,22 +614,17 @@ public void testEstimatedSizeWithStreamingBuffer() throws Exception {
   @Test
   public void testBigQueryQuerySourceEstimatedSize() throws Exception {
 
-    List<TableRow> data =
-        ImmutableList.of(
-            new TableRow().set("name", "A").set("number", 10L),
-            new TableRow().set("name", "B").set("number", 11L),
```

Comment:
Why did we have to significantly modify BigQueryIOReadTest (here and other places) ? It's better if we can keep existing tests and add another set of tests for storage API It's OK if this results in a little bit of code duplication (any large duplications can be moved to utility classes). If this PR significantly improves existing tests, please send that as a separate PR to be submitted before submitting this PR.

Reply:
https://github.com/apache/beam/pull/8061/commits/8493e349bede471e9af6198afce2cbcadd7a305f contains only the test refactoring, and explains the rationale for why the changes are needed. I can certainly refactor this into a separate PR if you think that's required.